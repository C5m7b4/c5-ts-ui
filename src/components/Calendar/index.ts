import Calendar, { CalendarProps } from './Calendar';

export { Calendar, CalendarProps };
export * from './Events';
