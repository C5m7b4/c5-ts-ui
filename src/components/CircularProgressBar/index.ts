import CircularProgressBar, {
  CircularProgressProps,
} from './CircularProgressBar';

export { CircularProgressBar, CircularProgressProps };
