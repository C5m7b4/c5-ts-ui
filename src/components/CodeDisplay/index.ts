import CodeDisplay, { CodeDisplayProps } from './CodeDisplay';
import { EditorSvgProps, clipboard } from './Clipboard';

export { CodeDisplay, CodeDisplayProps, EditorSvgProps, clipboard };
