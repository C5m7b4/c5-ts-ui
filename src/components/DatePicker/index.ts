import DatePicker, { DatePickerProps } from './DatePicker';
import DatePickerContents, {
  DatePickerContentProps,
} from './DatePickerContents';

export {
  DatePicker,
  DatePickerProps,
  DatePickerContentProps,
  DatePickerContents,
};
