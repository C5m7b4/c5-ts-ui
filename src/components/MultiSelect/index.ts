import MultiSelect, { MultiSelectProps } from './MultiSelect';
import MultiSelectPortal, { MultiSelectPortalProps } from './MultiSelectPortal';
export {
  MultiSelect,
  MultiSelectProps,
  MultiSelectPortal,
  MultiSelectPortalProps,
};
