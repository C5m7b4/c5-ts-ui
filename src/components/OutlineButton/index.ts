import OutlineButton, {
  OutlineButtonTypes,
  OutlineButtonProps,
} from './OutlineButton';

export { OutlineButtonTypes, OutlineButtonProps, OutlineButton };
