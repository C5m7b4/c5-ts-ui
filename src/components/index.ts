import {
  Button,
  NormalButtonType,
  NormalButtonSize,
  ButtonProps,
} from './Button';

export { Button, NormalButtonType, NormalButtonSize, ButtonProps };
