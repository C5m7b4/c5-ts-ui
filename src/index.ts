import { Add } from './math';

export { Add };
