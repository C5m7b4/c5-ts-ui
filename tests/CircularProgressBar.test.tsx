import React from 'react';
import { render } from '@testing-library/react';

import { CircularProgressBar } from '../src/components';

describe('CircularProgressBar tests', () => {
  test('should render correctly', () => {
    const { container } = render(<CircularProgressBar progress={80} />);
    expect(container).toMatchSnapshot();
  });

  test('should render with a score above 50', () => {
    const { container } = render(<CircularProgressBar progress={55} />);
    expect(container).toMatchSnapshot();
  });

  test('should render with a score below 50', () => {
    const { container } = render(<CircularProgressBar progress={45} />);
    expect(container).toMatchSnapshot();
  });

  test('should render with a score below 25', () => {
    const { container } = render(<CircularProgressBar progress={17} />);
    expect(container).toMatchSnapshot();
  });

  test('should render with a score below 75', () => {
    const { container } = render(<CircularProgressBar progress={71} />);
    expect(container).toMatchSnapshot();
  });

  test('should render with a score with 25', () => {
    const { container } = render(<CircularProgressBar progress={25} />);
    expect(container).toMatchSnapshot();
  });

  test('should handle dark mode', () => {
    const { container } = render(
      <CircularProgressBar progress={50} mode="dark" />
    );
    expect(container).toMatchSnapshot();
  });

  test('should handle dark mode', () => {
    const { container } = render(
      <CircularProgressBar progress={50} mode="light" />
    );
    expect(container).toMatchSnapshot();
  });

  test('should handle not showing progress', () => {
    const { container } = render(
      <CircularProgressBar progress={50} mode="light" showPercent={false} />
    );
    expect(container).toMatchSnapshot();
  });
});
