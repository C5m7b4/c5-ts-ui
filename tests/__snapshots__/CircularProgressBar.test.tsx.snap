// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircularProgressBar tests should handle dark mode 1`] = `
<div>
  <div>
    <svg
      class="circular-progress-bar"
      height="150"
      width="150"
    >
      <circle
        class="circular-progress-bar-bg"
        cx="75"
        cy="75"
        fill="none"
        r="67.5"
        stroke="#d9edfe"
        stroke-width="15"
      />
      <circle
        class="circulare-progress-bar-circle"
        cx="75"
        cy="75"
        fill="none"
        r="67.5"
        stroke="#daf542"
        stroke-dasharray="424.11500823462205"
        stroke-dashoffset="212.05750411731103"
        stroke-width="15"
        style="transition: stroke-dashoffset 850ms ease-in-out, stroke 500ms linear;"
      />
      <text
        class="circular-progress-bar-percentage"
        fill="#000"
        x="75"
        y="75"
      >
        50
        %
      </text>
    </svg>
  </div>
</div>
`;

exports[`CircularProgressBar tests should handle dark mode 2`] = `
<div>
  <div>
    <svg
      class="circular-progress-bar"
      height="150"
      width="150"
    >
      <circle
        class="circular-progress-bar-bg"
        cx="75"
        cy="75"
        fill="none"
        r="67.5"
        stroke="#d9edfe"
        stroke-width="15"
      />
      <circle
        class="circulare-progress-bar-circle"
        cx="75"
        cy="75"
        fill="none"
        r="67.5"
        stroke="#daf542"
        stroke-dasharray="424.11500823462205"
        stroke-dashoffset="212.05750411731103"
        stroke-width="15"
        style="transition: stroke-dashoffset 850ms ease-in-out, stroke 500ms linear;"
      />
      <text
        class="circular-progress-bar-percentage"
        fill="#fff"
        x="75"
        y="75"
      >
        50
        %
      </text>
    </svg>
  </div>
</div>
`;

exports[`CircularProgressBar tests should handle not showing progress 1`] = `
<div>
  <div>
    <svg
      class="circular-progress-bar"
      height="150"
      width="150"
    >
      <circle
        class="circular-progress-bar-bg"
        cx="75"
        cy="75"
        fill="none"
        r="67.5"
        stroke="#d9edfe"
        stroke-width="15"
      />
      <circle
        class="circulare-progress-bar-circle"
        cx="75"
        cy="75"
        fill="none"
        r="67.5"
        stroke="#daf542"
        stroke-dasharray="424.11500823462205"
        stroke-dashoffset="212.05750411731103"
        stroke-width="15"
        style="transition: stroke-dashoffset 850ms ease-in-out, stroke 500ms linear;"
      />
      <text
        class="circular-progress-bar-percentage"
        fill="#fff"
        x="75"
        y="75"
      >
        50
      </text>
    </svg>
  </div>
</div>
`;

exports[`CircularProgressBar tests should render correctly 1`] = `
<div>
  <div>
    <svg
      class="circular-progress-bar"
      height="150"
      width="150"
    >
      <circle
        class="circular-progress-bar-bg"
        cx="75"
        cy="75"
        fill="none"
        r="67.5"
        stroke="#d9edfe"
        stroke-width="15"
      />
      <circle
        class="circulare-progress-bar-circle"
        cx="75"
        cy="75"
        fill="none"
        r="67.5"
        stroke="#47e61c"
        stroke-dasharray="424.11500823462205"
        stroke-dashoffset="84.82300164692441"
        stroke-width="15"
        style="transition: stroke-dashoffset 850ms ease-in-out, stroke 500ms linear;"
      />
      <text
        class="circular-progress-bar-percentage"
        fill="#000"
        x="75"
        y="75"
      >
        80
        %
      </text>
    </svg>
  </div>
</div>
`;

exports[`CircularProgressBar tests should render with a score above 50 1`] = `
<div>
  <div>
    <svg
      class="circular-progress-bar"
      height="150"
      width="150"
    >
      <circle
        class="circular-progress-bar-bg"
        cx="75"
        cy="75"
        fill="none"
        r="67.5"
        stroke="#d9edfe"
        stroke-width="15"
      />
      <circle
        class="circulare-progress-bar-circle"
        cx="75"
        cy="75"
        fill="none"
        r="67.5"
        stroke="#daf542"
        stroke-dasharray="424.11500823462205"
        stroke-dashoffset="190.85175370557994"
        stroke-width="15"
        style="transition: stroke-dashoffset 850ms ease-in-out, stroke 500ms linear;"
      />
      <text
        class="circular-progress-bar-percentage"
        fill="#000"
        x="75"
        y="75"
      >
        55
        %
      </text>
    </svg>
  </div>
</div>
`;

exports[`CircularProgressBar tests should render with a score below 25 1`] = `
<div>
  <div>
    <svg
      class="circular-progress-bar"
      height="150"
      width="150"
    >
      <circle
        class="circular-progress-bar-bg"
        cx="75"
        cy="75"
        fill="none"
        r="67.5"
        stroke="#d9edfe"
        stroke-width="15"
      />
      <circle
        class="circulare-progress-bar-circle"
        cx="75"
        cy="75"
        fill="none"
        r="67.5"
        stroke="#f00e0e"
        stroke-dasharray="424.11500823462205"
        stroke-dashoffset="352.0154568347363"
        stroke-width="15"
        style="transition: stroke-dashoffset 850ms ease-in-out, stroke 500ms linear;"
      />
      <text
        class="circular-progress-bar-percentage"
        fill="#000"
        x="75"
        y="75"
      >
        17
        %
      </text>
    </svg>
  </div>
</div>
`;

exports[`CircularProgressBar tests should render with a score below 50 1`] = `
<div>
  <div>
    <svg
      class="circular-progress-bar"
      height="150"
      width="150"
    >
      <circle
        class="circular-progress-bar-bg"
        cx="75"
        cy="75"
        fill="none"
        r="67.5"
        stroke="#d9edfe"
        stroke-width="15"
      />
      <circle
        class="circulare-progress-bar-circle"
        cx="75"
        cy="75"
        fill="none"
        r="67.5"
        stroke="#f59b33"
        stroke-dasharray="424.11500823462205"
        stroke-dashoffset="233.26325452904214"
        stroke-width="15"
        style="transition: stroke-dashoffset 850ms ease-in-out, stroke 500ms linear;"
      />
      <text
        class="circular-progress-bar-percentage"
        fill="#000"
        x="75"
        y="75"
      >
        45
        %
      </text>
    </svg>
  </div>
</div>
`;

exports[`CircularProgressBar tests should render with a score below 75 1`] = `
<div>
  <div>
    <svg
      class="circular-progress-bar"
      height="150"
      width="150"
    >
      <circle
        class="circular-progress-bar-bg"
        cx="75"
        cy="75"
        fill="none"
        r="67.5"
        stroke="#d9edfe"
        stroke-width="15"
      />
      <circle
        class="circulare-progress-bar-circle"
        cx="75"
        cy="75"
        fill="none"
        r="67.5"
        stroke="#daf542"
        stroke-dasharray="424.11500823462205"
        stroke-dashoffset="122.99335238804039"
        stroke-width="15"
        style="transition: stroke-dashoffset 850ms ease-in-out, stroke 500ms linear;"
      />
      <text
        class="circular-progress-bar-percentage"
        fill="#000"
        x="75"
        y="75"
      >
        71
        %
      </text>
    </svg>
  </div>
</div>
`;

exports[`CircularProgressBar tests should render with a score with 25 1`] = `
<div>
  <div>
    <svg
      class="circular-progress-bar"
      height="150"
      width="150"
    >
      <circle
        class="circular-progress-bar-bg"
        cx="75"
        cy="75"
        fill="none"
        r="67.5"
        stroke="#d9edfe"
        stroke-width="15"
      />
      <circle
        class="circulare-progress-bar-circle"
        cx="75"
        cy="75"
        fill="none"
        r="67.5"
        stroke="#f59b33"
        stroke-dasharray="424.11500823462205"
        stroke-dashoffset="318.0862561759665"
        stroke-width="15"
        style="transition: stroke-dashoffset 850ms ease-in-out, stroke 500ms linear;"
      />
      <text
        class="circular-progress-bar-percentage"
        fill="#000"
        x="75"
        y="75"
      >
        25
        %
      </text>
    </svg>
  </div>
</div>
`;
